src/plugins/intel_gpu/src/kernel_selector/kernels/gemm/gemm_kernel_tiled_opt.cpp

```diff
@@ -109,7 +109,8 @@ GemmKernelTiledOpt::GemmTuningData GemmKernelTiledOpt::SetTuningParams(const gem
         bool output_ndim_transposed = (params.output_order.size() > 0 && (params.output_order.back() != (static_cast<int>(params.output_order.size()) - 1)));
         if ((params.transpose_input0 == 0 /*X_LAST*/) && (params.transpose_input1 == 0 /*X_LAST*/ || params.transpose_input1 == 1 /*Y_LAST*/)
             && (!params.indirect_input0 && !params.inputs[0].has_dynamic_pad())
-            && (!output_ndim_transposed || params.fused_ops.empty())) {
+            && (!output_ndim_transposed || params.fused_ops.empty())
+            && params.engineInfo.deviceType == dev_type::integrated_gpu) {
```

Comment:
please use `!supports_immad` instead of `integrated_gpu` because these are might not be the same.